grant-project/grant-monorepo
# Functioning proposal teams (pt 2 - the views)

### What This Does

* Updates `TeamBlock` and `ProposalCard` to render the team, rather than the trustees.
* Split out `UserRow` into `AddressRow` and `UserRow` components. AddressRow always renders address / identicon and optional secondary info. User row renders username, title, and either avatar, identicon, or placeholder image depending on what's available.
    * Also added a UserAvatar component that handles the image fallback, used in create flow too
    * Added a storybook entry for UserRow
* Transforms team member objects to and from the API to be one uniform `TeamMember` type.
* Add social medias to example proposal

### Things That Could Be Improved

* The `User` type and `TeamMember` type should probably be unified. This will likely be done at some point as a part of adding auth when we start using full users in a lot of places.
* There is no longer a place where trustees are listed. We may want an additional tab that shows technical data about the proposal?

### Steps to Test

* Create a proposal with a team
* Ensure the team block shows the team you added
* Ensure the proposal card shows the team you added

^ Note that I did not test with legacy data, and errors may occur on proposals that don't have teams.

## Screenshots

### Team Block

<img width="327" alt="screen shot 2018-09-27 at 1 28 45 pm" src="https://user-images.githubusercontent.com/649992/46163448-49c00900-c259-11e8-8266-3759d3cea829.png">


### Proposal Card

<img width="361" alt="screen shot 2018-09-27 at 1 28 39 pm" src="https://user-images.githubusercontent.com/649992/46163453-4cbaf980-c259-11e8-8bc3-9ec5f6b5b2eb.png">
